### PR TITLE
isuse/5853

### LINKED ISSUE #5853: API authentication is strongly-tied to global $wp_query
I'm reverse engineering the `EDD_API::validate_request()` method so I can check API keys for a POST request to WP REST API. I'm cut and pasting the key hash check, would be better if that was separated out.

I'll put in a PR.

### PR DESCRIPTION
**Pull request by sunnyratilal**
Fixes #5853

### CODE REVIEW

File: includes/api/class-edd-api.php [hunk 1]
```diff
@@ -329,11 +329,13 @@ private function validate_request() {
 				$secret = $this->get_user_secret_key( $user );
 				$public = urldecode( $wp_query->query_vars['key'] );
 
-				if ( hash_equals( md5( $secret . $public ), $token ) ) {
+				$valid = $this->check_keys( $secret, $public, $token );
+				if ( $valid ) {
 					$this->is_valid_request = true;
-				} else {
+				} else{
```
Comment (pippinsplugins) [inline includes/api/class-edd-api.php]: Missed a space here.

File: includes/api/class-edd-api.php [hunk 2]
```diff
@@ -2267,4 +2269,25 @@ public function log_requests() {
...
+	 * @since  2.8.2
+	 *
+	 * @param string $secret Secret key
+	 * @param string $public Public key
+	 * @param string $token Token used in API request
+	 *
+	 * @return bool
+	 */
+	public function check_keys( $secret, $public, $token ){
+		if ( hash_equals( md5( $secret . $public ), $token ) ) {
```
Comment (pippinsplugins) [inline includes/api/class-edd-api.php]: Can we go ahead and make this a single line? `return hash_equals( md5( $secret . $public ), $token )`.

**Comment by sunnyratilal** [comment]
@pippinsplugins PR updated

**Comment by pippinsplugins** [comment]
@sunnyratilal I'm not seeing the updated return in `check_keys`.

**Comment by sunnyratilal** [comment]
@pippinsplugins Ah, I missed the line comment you made. PR updated

**Pull request merged**